x-pack/plugins/ml/public/application/components/job_spaces_selector/spaces_selectors.tsx

```diff
@@ -0,0 +1,204 @@
+    setSelectedSpaceIds(ids);
+  }
+
+  const isGlobalControlChecked = selectedSpaceIds.includes(ALL_SPACES_ID);
+
+  const options = allSpaces.map<SpaceOption>((space) => {
+    return {
+      label: space.name,
+      prepend: <SpaceAvatar space={space} size={'s'} />,
+      checked: selectedSpaceIds.includes(space.id) ? 'on' : undefined,
+      disabled: canEditSpaces === false,
+      ['data-space-id']: space.id,
+      ['data-test-subj']: `cts-space-selector-row-${space.id}`,
+    };
+  });
+
+  const shareToAllSpaces = {
+    id: 'shareToAllSpaces',
+    title: i18n.translate('xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.title', {
+      defaultMessage: 'All spaces',
+    }),
+    text: i18n.translate('xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.text', {
+      defaultMessage: 'Make job available in all current and future spaces.',
+    }),
+    ...(!canShareToAllSpaces && {
+      tooltip: isGlobalControlChecked
+        ? i18n.translate(
+            'xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.cannotUncheckTooltip',
+            { defaultMessage: 'You need additional privileges to change this option.' }
+          )
+        : i18n.translate(
+            'xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.cannotCheckTooltip',
+            { defaultMessage: 'You need additional privileges to use this option.' }
+          ),
+    }),
+    disabled: !canShareToAllSpaces,
+  };
+
+  const shareToExplicitSpaces = {
+    id: 'shareToExplicitSpaces',
+    title: i18n.translate('xpack.ml.management.spacesSelectorFlyout.shareToExplicitSpaces.title', {
+      defaultMessage: 'Select spaces',
+    }),
+    text: i18n.translate('xpack.ml.management.spacesSelectorFlyout.shareToExplicitSpaces.text', {
+      defaultMessage: 'Make job available in selected spaces only.',
+    }),
+    disabled: !canShareToAllSpaces && isGlobalControlChecked,
+  };
+
+  return (
+    <>
+      {canEditSpaces === false && <CannotEditCallout jobId={jobId} />}
+      <EuiFormFieldset>
+        <EuiCheckableCard
+          id={shareToExplicitSpaces.id}
+          label={createLabel(shareToExplicitSpaces)}
+          checked={!isGlobalControlChecked}
+          onChange={() => toggleShareOption(false)}
+          disabled={shareToExplicitSpaces.disabled}
+        >
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.management.spacesSelectorFlyout.selectSpacesLabel"
+                defaultMessage="Select spaces"
+              />
+            }
+            fullWidth
+          >
+            <EuiSelectable
+              options={options}
+              onChange={(newOptions) => updateSelectedSpaces(newOptions as SpaceOption[])}
+              listProps={{
+                bordered: true,
+                rowHeight: 40,
+                className: 'spcCopyToSpace__spacesList',
```

Comment:
This relies on the SCSS code that is in the Spaces plugin, which is a brittle dependency.

I would recommend renaming this class and making a separate SCSS file for it in this plugin.

Reply:
thanks for spotting this, it was left in by accident.

Reply:
fixed in https://github.com/elastic/kibana/pull/83320/commits/b4300cd9d11de9816e71d18a2a3695d36918d111